packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -130,12 +138,39 @@ export class SOQLEditorInstance {
         });
         break;
       }
+      case MessageType.RUN_SOQL_QUERY: {
+        this.handleRunQuery().catch(() => {
+          channelService.appendLine(
+            `An error occurred while running the SOQL query.`
+          );
+        });
+        break;
+      }
       default: {
         console.log('message type is not supported');
       }
     }
   }
 
+  protected handleRunQuery(): Promise<void> {
+    const queryText = this.document.getText();
```

Comment:
I think this code that runs the query and opens in the webview should be moved to it's own class so it can be accessed in a general way, since it will be useful outside the context of the SOQL Builder editor. If not moved in this PR, a comment to that effect should be added for future movement.

Reply:
There are two classes `QueryRunner` and `QueryDataViewService'. The runner is really a wrapper for the connection that also cleans the results for display. The query view service can be used to generate a table if given the data etc.
Is there something more general that your are looking for?
Eventually our extension might want to export some of these services to be used by other extensions.

Reply:
It's probably ok. I was thinking that the run-and-show code that you're handling here would be valuable in some independent class... RunAndShowQuery or some such thing. We can always circle back if we need to.

Reply:
I think we should soon extract the calls to the Salesforce platform from `soqlEditorInstance` into a separate module/class, including the calls to retrieve sobjects metadata, run query, etc.